### ISSUE TITLE
I no longer get updates in Home Assistant 2023.5.3

### ISSUE THREAD

**Issue by pd5wl** [xirixiz/homeassistant-afvalwijzer#240]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/xirixiz/homeassistant-afvalwijzer/blob/master/.github/SUPPORT.md)

### Description

All entities say unavaileble as of the update to 2023.5.3.

After doing the update all these entities went "unavaileble". No configs where changed and it's the latetst version.

### Expected behaviour

Show waste information. 

### Actual behaviour

On the dashboard, in my automations and in the entities list it says "unavaileble" and the following warning. "This entity is no longer being provided by the afvalwijzer integration. If the entity is no longer in use, delete it in settings."
![Knipsel](https://github.com/xirixiz/homeassistant-afvalwijzer/assets/11857240/00af78e5-a4f9-4a40-ab5e-7beefa0757c4)
![Knipsel1](https://github.com/xirixiz/homeassistant-afvalwijzer/assets/11857240/5c9881e1-8e08-471f-8553-0ad596012119)

### Version

* HASS version:

Home Assistant 2023.5.3
Supervisor 2023.04.1
Operating System 10.1
Frontend 20230503.3 - latest

* Component version:
"version": "2023.04.02"

* Useful system info:
VM on a Synology

### HASS logs

Sadly no information what so ever:
![Knipsel2](https://github.com/xirixiz/homeassistant-afvalwijzer/assets/11857240/65858fc5-be1e-4dc9-a30f-0006cae9297e)
![Knipsel3](https://github.com/xirixiz/homeassistant-afvalwijzer/assets/11857240/6eca3a42-a7d0-456d-8434-f7f9ddbc2118)

**Comment by basvanrooten** [comment]
In addition, I believe that on 2023.5.2 the integration is also not working. I also do not have any logs, but it seems to have stopped functioning around midnight on may 11. I did not update, reboot or do anything at that time.

![image](https://github.com/xirixiz/homeassistant-afvalwijzer/assets/33634284/f33f39fd-69b2-43c0-b884-ee654d7f4b26)

**Comment by xirixiz** [comment]
Hi, it seems more related of a provider switch in the region you are living.
What are your postal codes so I can validate some things?
Also, can you check the logfiles for errors and share those with me?

**Comment by basvanrooten** [comment]
Unfortunately I cannot find any relevant logging. Because of the Github guidelines I cannot share **my** postal code, so here's one from a street really close to me. 4854 ET.

Thank you for taking the time to debug, really appreciate the work!

**Comment by xirixiz** [comment]
Hi sure, no prob. I will have a look later Today. 

I`m just curious, "cannot share my postal code" because of GitHub guidelines? I've never heard of it actually. Where can I find this? I ask ppl all the time for their postal code in case of issues.

**Comment by basvanrooten** [comment]
> I`m just curious, "cannot share my postal code" because of GitHub guidelines? I've never heard of it actually. Where can I find this? I ask ppl all the time for their postal code in case of issues.

I wasn't sure if I was able to do that so I looked it up in the [Community Code of Conduct](https://docs.github.com/en/site-policy/github-terms/github-community-code-of-conduct#best-practices-for-maintaining-a-strong-community).

**Comment by xirixiz** [comment]
I already checked, but the data is being retrieved nicely. I really need some logfiles to be able to identify the issue.
Maybe mijnafvalwijzer was having some issues. Could you restart HASS once again?


```
-> python3 -m afvalwijzer.tests.test_module

Data collected from: mijnafvalwijzer with postcal code: 4854ET

{'gft': datetime.datetime(2023, 5, 23, 0, 0), 'papier': datetime.datetime(2023, 5, 25, 0, 0), 'plastic': datetime.datetime(2023, 5, 31, 0, 0)}
{'gft': datetime.datetime(2023, 5, 23, 0, 0), 'papier': datetime.datetime(2023, 5, 25, 0, 0), 'plastic': datetime.datetime(2023, 5, 31, 0, 0)}
{'next_date': datetime.datetime(2023, 5, 23, 0, 0), 'next_in_days': 5, 'next_type': 'gft', 'today': 'geen', 'tomorrow': 'geen', 'day_after_tomorrow': 'geen'}
['gft', 'papier', 'plastic']
['day_after_tomorrow', 'next_date', 'next_in_days', 'next_type', 'today', 'tomorrow']
```

Als make sure that your settings are valid @basvanrooten. Something like this: 
```
provider = "mijnafvalwijzer"
postal_code = "4854ET"
street_number = "30"
```

**Comment by xirixiz** [comment]
> > I`m just curious, "cannot share my postal code" because of GitHub guidelines? I've never heard of it actually. Where can I find this? I ask ppl all the time for their postal code in case of issues.
> 
> I wasn't sure if I was able to do that so I looked it up in the [Community Code of Conduct](https://docs.github.com/en/site-policy/github-terms/github-community-code-of-conduct#best-practices-for-maintaining-a-strong-community).

Thanks for sharing! I wasn't aware, so next time I`ll take this into account.

**Comment by pd5wl** [comment]
Sorry for the later reply, work caught up on me.

sensor:
  - platform: afvalwijzer
    provider: mijnafvalwijzer
    postal_code: 3941RK
    street_number:  50
    suffix: 'B'
    exclude_pickup_today: flase
    default_label: geen

My afwijzer app works fine.

**Comment by xirixiz** [comment]
@pd5wl, no problem ofcourse 😄 .

Just checked, and I`m also retrieving data for your postal code:

```
-> python3 -m afvalwijzer.tests.test_module


Data collected from: mijnafvalwijzer with postcal code: 3941RK


{'milieustraat_gesloten': datetime.datetime(2023, 5, 18, 0, 0), 'restafval': datetime.datetime(2023, 5, 19, 0, 0), 'gft': datetime.datetime(2023, 5, 31, 0, 0), 'pmd': datetime.datetime(2023, 6, 2, 0, 0), 'papier': datetime.datetime(2023, 6, 10, 0, 0), 'kca': datetime.datetime(2023, 9, 16, 0, 0), 'takken': datetime.datetime(2023, 10, 18, 0, 0), 'textiel': datetime.datetime(2023, 11, 3, 0, 0)}
{'restafval': datetime.datetime(2023, 5, 19, 0, 0), 'milieustraat_gesloten': datetime.datetime(2023, 5, 29, 0, 0), 'gft': datetime.datetime(2023, 5, 31, 0, 0), 'pmd': datetime.datetime(2023, 6, 2, 0, 0), 'papier': datetime.datetime(2023, 6, 10, 0, 0), 'kca': datetime.datetime(2023, 9, 16, 0, 0), 'takken': datetime.datetime(2023, 10, 18, 0, 0), 'textiel': datetime.datetime(2023, 11, 3, 0, 0)}
{'next_date': datetime.datetime(2023, 5, 19, 0, 0), 'next_in_days': 1, 'next_type': 'restafval', 'today': 'milieustraat_gesloten', 'tomorrow': 'restafval', 'day_after_tomorrow': 'geen'}
['gft', 'kca', 'milieustraat_gesloten', 'papier', 'pmd', 'restafval', 'takken', 'textiel']
['day_after_tomorrow', 'next_date', 'next_in_days', 'next_type', 'today', 'tomorrow']
```

Can you restart HASS once again, maybe there was an issue Yesterday.

**Comment by pd5wl** [comment]
Waiting on 2 updates to finish and then i'll do a restart.

**Comment by pd5wl** [comment]
Not shure i got a Error in the logs, but i can't link it to afvalwijzer. wich still say unavaileble. I haven't seen this before.

Logger: homeassistant.core
Source: helpers/integration_platform.py:126
First occurred: 09:13:28 (1 occurrences)
Last logged: 09:13:28

Error in event filter
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/core.py", line 1011, in async_fire
    if not event_filter(event):
  File "/usr/src/homeassistant/homeassistant/helpers/integration_platform.py", line 126, in _async_component_loaded_filter
    return "." not in event.data[ATTR_COMPONENT]
KeyError: 'component'

**Comment by xirixiz** [comment]
It seems to be related to another issue. This parts loads the custom components, but it's not clear which component is causing this issue. Since there aren't a lot of issues reported here, I assume it's related to another component you both might be using.

I would advice to turn on debug loggin in HASS itself, and maybe as well for this component, to get more insight in what is causing the issue exactly. Please share if you have additional info. Currently I cannot do more or provide you any additional support without more detailed info.

**Comment by pd5wl** [comment]
I set the logging to debug for all. The only other custom_comp. i have is KNMI. wich works fine. So here goes another reboot.

**Comment by pd5wl** [comment]
Ok found some logs: 

2023-05-18 09:39:53.486 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration afvalwijzer which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
Voor alle entities:
2023-05-18 09:40:39.084 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: entity_id=sensor.afvalwijzer_gft, old_state=None, new_state=<state sensor.afvalwijzer_gft=unavailable; restored=True, icon=mdi:recycle, friendly_name=afvalwijzer gft, supported_features=0 @ 2023-05-18T09:40:39.084262+02:00>>
2023-05-18 09:40:42.661 DEBUG (Recorder) [homeassistant.components.recorder.core] Processing task: EventTask(event=<Event state_changed[L]: entity_id=sensor.afvalwijzer_gft, old_state=None, new_state=<state sensor.afvalwijzer_gft=unavailable; restored=True, icon=mdi:recycle, friendly_name=afvalwijzer gft, supported_features=0 @ 2023-05-18T09:40:39.084262+02:00>>)

Posting just one.

There is a lot in the

**Comment by pd5wl** [comment]
I just removed the other custom_comp. Results where the same. unvaileble.

**Comment by xirixiz** [comment]
Kan het misschien zijn dat je meerdere GFT entries hebt? Kan je eens door de afvalwijzer entities lopen?
Developer Tools -> States. Mogelijk dat er een tweede afvalwijzer_gft entry is die wel de juiste info bevat? Soms gebeurd dat namelijk helaas. Zoiets zeg maar:

![image](https://github.com/xirixiz/homeassistant-afvalwijzer/assets/11210639/d4439bc7-9106-44d4-80bb-9a91e99d90d6)

**Comment by pd5wl** [comment]
Nope, don't show.. It's so wierd.  @basvanrooten Did it came back with you? Or stil the same? 

![Knipsel100](https://github.com/xirixiz/homeassistant-afvalwijzer/assets/11857240/5813d4ae-974a-4600-8c9f-c4971f1264db)

**Comment by basvanrooten** [comment]
Nope, exact same behavior as @pd5wl. Also no duplicate entities, just the unavailable ones, no logging available.

**Comment by pd5wl** [comment]
I've been trying to get any log data but even with the debug setting there is nothing. That make me wonder, does the code run? I tried running python from the terminal, but it could not find python. 

What is the easiest way to test this.

**Comment by pd5wl** [comment]
Ok I tried the test command u used:

python3 -m afvalwijzer.tests.test_module

But i didnt get any results, but i got this
![Knipsel99](https://github.com/xirixiz/homeassistant-afvalwijzer/assets/11857240/20441fba-0a1b-4d46-9d29-eaa24ae1f88a)

**Comment by pd5wl** [comment]
stuk at this point... 

![Knipsel98](https://github.com/xirixiz/homeassistant-afvalwijzer/assets/11857240/112c3b6c-3252-4275-a861-10400dffbee8)

**Comment by xirixiz** [comment]
Nice job exploring it yourself, I like it ;).
The top line says you're missing the request module for python. You can install this by using pip.

```
pip3 install requests --user
```

If pip3 doesn't exist, you can try ```pip``` instead.

**Comment by pd5wl** [comment]
requests installed.

Test is now working.

![Knipsel_werkt](https://github.com/xirixiz/homeassistant-afvalwijzer/assets/11857240/74400bb5-38de-4944-b955-e869e8bb800e)

Lets see if the rest works as well.

**Comment by pd5wl** [comment]
Ok. Next step. I did the following:

Tests ran great, enities didnt do anything. 

Removed afvakwijzer from custom_temp. Restarterd HA. Removed old afval.* enities.
Restarted HA. cheacked.. afvalwijzer is gone.
Reinstalled afval wijzer in cutom_tmp. And now it's getting somewhere..... Well not really. But no entities where made. 
This adds up to the earlier notification i had and i didn't screenshot. 

So basicly, tests run fine. Deleted the folder, cleared the entities, reboot,checked => gone, reinstalled, reboot, test works fine but no entities. 

This all happend after the last update to 2023.5.3. Also no logging.

is there any thing i miss here?

**Comment by xirixiz** [comment]
Thanks for checking some things! It's very strange behaviour. What I don't know about is custom_t(e)mp. I only have ```<hass>/config/custom_components``` where I have to put the custom integration. Do you have HACS installed? I update the component via HACS, and then also the component is placed in custom_components. 

The way you describe the problem feels like the custom_component is not found/initiated at all. Can you check this maybe?

**Comment by pd5wl** [comment]
My bad, i meant "/custom_components"

I Also have KNMI in the /custom/comp. This works fine.
Ok Installed HACS, reinstalled the add-on. same results.
![Knipsel2](https://github.com/xirixiz/homeassistant-afvalwijzer/assets/11857240/1ba87364-fe42-435c-a7d8-50982de8402b)
![Knipsel](https://github.com/xirixiz/homeassistant-afvalwijzer/assets/11857240/80bc69ab-ae77-45ef-88bc-bfd6726d2d7c)

Any idea's where HA caches entities? Maybe i need to look in to that.

**Comment by pd5wl** [comment]
Can be closed. Somehow i broke it and it wont let me reinstall. I found an other add-on that does the same. Put it in and worked.
I would have digget further, but as there is no debug info entering my logs it's a lost case for me. 

Sorry and thank you for the support, again it aint your add-on. It's something in my HASS

**Comment by xirixiz** [comment]
No problem ofcourse. It's really weird that you have this weird behaviour. It could be that the database might be corrupt or something, so it could be causing issues in the future. Anyway, the required info you need is at least available now 😄.

Thanks again for also checking things yourself.

**Comment by pd5wl** [comment]
I am looking in to the DB of HA. But it's so wierd that it doesn't seem to load at all, hence the missing debug logs.
If I might find something, ill will let you know.
